tools/congestion-model/src/workload/all_for_one.rs

```diff
@@ -29,45 +51,122 @@ impl Producer for AllForOneProducer {
     }
 
-    fn receipt_to_shard_0(&self, shards: &[ShardId]) -> ReceiptDefinition {
+    /// Transaction with a chain of three receipts, the last one spending the
+    /// most gas. The last receipt always has shard 0 as receiver.
+    fn produce_three_hops_tx(&self, shards: &[ShardId], tx: &mut TransactionBuilder) {
+        let my_index = shards
+            .iter()
+            .position(|&id| id == tx.sender_shard())
+            .expect("sender must be in shards");
+        let hop_1 = shards[(my_index + 1) % shards.len()];
+        let hop_2 = shards[(my_index + 2) % shards.len()];
+
+        let light_receipt_1 = self.light_receipt(hop_1, 0);
+        let light_receipt_2 = self.light_receipt(hop_2, 1);
+        let heavy_receipt = self.receipt_to_shard_0(shards, 2);
+
+        let first = tx.add_first_receipt(light_receipt_1, self.conversion_gas);
+        let second = tx.new_outgoing_receipt(first, light_receipt_2);
+        let third = tx.new_outgoing_receipt(second, heavy_receipt);
+
+        tx.new_outgoing_receipt(first, utils::refund_receipt(tx.sender_shard()));
+        tx.new_outgoing_receipt(second, utils::refund_receipt(tx.sender_shard()));
+        tx.new_outgoing_receipt(third, utils::refund_receipt(tx.sender_shard()));
+    }
+
+    fn receipt_to_shard_0(&self, shards: &[ShardId], prior_hops: u64) -> ReceiptDefinition {
         ReceiptDefinition {
             receiver: shards[0],
             size: self.receipt_size,
-            attached_gas: self.attached_gas,
-            execution_gas: self.attached_gas,
+            attached_gas: self.attached_gas - self.last_execution_gas * prior_hops,
+            execution_gas: self.last_execution_gas,
+        }
+    }
+
+    fn light_receipt(&self, shard: ShardId, prior_hops: u64) -> ReceiptDefinition {
+        ReceiptDefinition {
+            receiver: shard,
+            size: self.receipt_size,
+            attached_gas: self.attached_gas - self.last_execution_gas * prior_hops,
+            execution_gas: self.light_execution_gas,
+        }
+    }
+
+    pub fn one_hop_only() -> Self {
+        Self {
+            enable_one_hop: true,
+            enable_two_hops: false,
+            enable_three_hops: false,
+            ..Default::default()
         }
     }
 }
 
 impl Default for AllForOneProducer {
     fn default() -> Self {
         // for this simple scenario, as a default, use 1kb sized receipts with 300 Tgas attached that
-        // end up actually executing 5 TGas + 100 TGas.
+        // end up actually executing 5 TGas + 100 TGas (+ optionally 2x 10 Tgas).
         Self {
             receipt_size: 1024,
             attached_gas: 300 * TGAS,
-            execution_gas: 100 * TGAS,
+            light_execution_gas: 10 * TGAS,
+            last_execution_gas: 100 * TGAS,
             conversion_gas: 5 * TGAS,
-            messages_per_round: 20,
+            // ideally a number divisible by 3
+            messages_per_round: 42,
```

Comment:
Because the execution gas was set to 300Tgas instead of 100 Tgas as intended, I had to increase the total number here to reach congestion again.
Also make it divisible by three to have three evenly sized thirds.